src/basic/hostname-util.c

```diff
@@ -172,3 +172,36 @@ int get_pretty_hostname(char **ret) {
         *ret = TAKE_PTR(n);
         return 0;
 }
+
+int split_user_at_host(const char *s, char **ret_user, char **ret_host) {
```

Comment:
Maybe better to check if the user name and host name are valid?

Reply:
We actually carry checks in sd-bus, so shouldn't be that much of a hassle. I have that in #32999 and might try to revive it after this PR.

Reply:
Oh, dang, I totally forgot that #32999 exists, even though I already reviewed that. Sorry for the duplication. If I had realized that I wouldn't have touched this. Sorry! I was just going through the milestone, and didn't see you already started working n it.